src/main/java/com/project/trav/service/TicketService.java

```diff
@@ -17,52 +20,69 @@
-        return ticketRepository.findById(id).orElseThrow(()->{
-            throw new EntityNotFoundByIdException(NOT_FOUND_ERROR);
-        });
+  public TicketDto getTicket(Long id) {
+    return ticketMapper.toTicketDto(findByIdTicket(id));
+  }
+  public void addTicket(TicketDto ticketDto) {
+    validPlace(ticketDto);
+    ticketRepository.save(ticketMapper.toTicket(ticketDto));
+  }
 
+  public void deleteTicket(Long id) {
+    findByIdTicket(id);
+    ticketRepository.deleteById(id);
+  }
+
+  public void updateTicket(TicketDto ticketDto, Long id) {
+    Ticket oldTicket = findByIdTicket(id);
+    validPlace(ticketDto);
+    ticketRepository.save(oldTicket.setId(id).setUserId(ticketDto.getUserId())
+        .setPlace(ticketDto.getPlace()).setPlaceClass(ticketDto.getPlaceClass()).setCost(ticketDto.getCost())
+        .setTicketStatus(ticketDto.getTicketStatus()).setRaces(new Race().setId(ticketDto.getId())));
+  }
+
+  public void buyTicket(Long ticketId, String username) {
+    prepareTicket(TicketStatus.BOUGHT, ticketId, username);
+  }
+
+  public void bookTicket(Long ticketId, String username) {
+    prepareTicket(TicketStatus.BOOKED, ticketId, username);
+  }
+  private Ticket findByIdTicket(Long id) {
+    return ticketRepository.findById(id).orElseThrow(() -> {
+      throw new EntityNotFoundByIdException(NOT_FOUND_ERROR);
+    });
+
+  }
+  private void prepareTicket(TicketStatus ticketStatus, Long ticketId, String username) {
+    User user = userRepository.findByLogin(username).orElseThrow(() ->
+        new EntityNotFoundByIdException("User was not found"));
+    Ticket ticket = findByIdTicket(ticketId);
+    if (!ticket.getTicketStatus().equals(TicketStatus.AVAILABLE)) {
+      throw new TicketReservingException("Ticket is not available");
     }
-    private void existByIdTicket(Long id){
-        if(!ticketRepository.existsById(id)){
-            throw new EntityNotFoundByIdException(NOT_FOUND_ERROR);
-        }
-    }
-    private void prepareTicket(TicketStatus ticketStatus,Long ticketId,String username){
-        User user = userRepository.findByLogin(username).orElseThrow(()->
-            new EntityNotFoundByIdException("User was not found"));
-        Ticket ticket = findByIdTicket(ticketId);
-        if (!ticket.getTicketStatus().equals(TicketStatus.AVAILABLE)){
-            throw new TicketReservingException("Ticket is not available");
-        }
-        ticket.setUserId(user.getId());
-        ticket.setTicketStatus(ticketStatus);
-        updateTicket(ticketMapper.toTicketDto(ticket),ticketId);
-    }
+    ticket.setUserId(user.getId());
+    ticket.setTicketStatus(ticketStatus);
+    updateTicket(ticketMapper.toTicketDto(ticket), ticketId);
+  }
+  private void validPlace(TicketDto ticketDto) {
+    Race race = raceRepository.findById(ticketDto.getRacesDto().getId()).orElseThrow(() ->
+        new EntityNotFoundByIdException("Race was not fond"));
+    List<Ticket> ticketList = race.getTickets();
```

Comment:
I guess you can try to use only one query: 
 Optional<Ticket> findByPlaceAndRaces_RaceNumber(String place, String raceNumber);

But maybe you also need to add 
  @ToString.Exclude
  @EqualsAndHashCode.Exclude
behind races in Ticket.java

Please investigate this